types/constants.ts

```diff
@@ -1 +1,90 @@
-export const ARNS_NAME_REGEX = new RegExp('^(?!-)[a-z0-9-s+]{1,32}(?<!-)$');
+export const ARNS_NAME_REGEX = new RegExp('^(?!-)[a-zA-Z0-9-s+]{1,32}(?<!-)$');
+export const NAME_PRICES: { [x: number]: number } = {
+  1: 4218750000,
+  2: 1406250000,
+  3: 468750000,
+  4: 156250000,
+  5: 62500000,
+  6: 25000000,
+  7: 10000000,
+  8: 5000000,
+  9: 1000000,
+  10: 500000,
+  11: 450000,
+  12: 400000,
+  13: 350000,
+  14: 300000,
+  15: 250000,
+  16: 200000,
+  17: 175000,
+  18: 150000,
+  19: 125000,
+  20: 100000,
+  21: 100000,
+  22: 100000,
+  23: 100000,
+  24: 100000,
+  25: 100000,
+  26: 100000,
+  27: 100000,
+  28: 100000,
+  29: 100000,
+  30: 100000,
+  31: 100000,
+  32: 100000,
+};
```

Comment:
can't we get this from the global arns contract state?

Reply:
yup. But that only has prices for up to 20 chars. Will need to add some logic for that - we could actually also implement a regEx updater that looks at how many prices their are in the contract and limits the regex to that. Thoughs?

Reply:
20-32 is the same price, so if the length is greater than 20, use that price